src/runtime/schema.ts

```diff
@@ -57,10 +57,66 @@ export class Schema {
       }
     }
     if (options.description) {
+      // The descriptions should be passed ready for assignment into this.description.
+      // TODO(cypher1): Refactor the schema construction code to do this rearrangement at the call site.
       options.description.description.forEach(desc => this.description[desc.name] = desc.pattern || desc.patterns[0]);
     }
   }
 
+  private forEachRefinement<T, V>(func: (val: V) => T) {
+    const types = [this, ...Object.values(this.fields)];
+    types.forEach(type => {
+      if (type.refinement) {
+        func(type.refinement);
+      }
+    });
```

Comment:
Since you're already using a functional style forEach, you could filter as well:

`types.filter(t => t.refinement).forEach(t => func(t.refinement));`